pkg/connector/service.go

```diff
@@ -102,6 +102,7 @@ func (s *Service) Create(ctx context.Context, id string, t Type, cfg Config) (Co
 		return nil, cerrors.Errorf("could not init connector: %w", err)
 	}
 
+	conn.SetProvisionedBy(p)
```

Comment:
If we remove the method we would need to pass `ProvisionType` to the builder when the connector is created in `Builder.Build`.